digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java

```diff
@@ -549,6 +556,28 @@ public RestProject putProject(@QueryParam("project") String name, @QueryParam("r
                     if (md5Count.getCount() != contentLength) {
                         throw new IllegalArgumentException("Content-Length header doesn't match with uploaded data size");
                     }
+                    WorkflowDefinitionList defs = meta.getWorkflowList();
+                    for (WorkflowDefinition def : defs.get()) {
+                        Workflow wf = compiler.compile(def.getName(), def.getConfig());
+
+                        // validate workflow and schedule
+//                        Set<String> required = new HashSet<>();
+                        for (WorkflowTask task : wf.getTasks()) {
+                            // raise an exception if task doesn't valid.
+                            task.getConfig();
+//                            String require = config.getOptional("require>", String.class).orNull();
+//                            if (require != null && required.add(require)) {
+//                                f.ln("  -> %s", require);
+//                            }
+                        }
+                        Revision rev = Revision.builderFromArchive("check", meta, getUserInfo())
+                                .archiveType(ArchiveType.NONE)
+                                .build();
+                        // raise an exception if "schedule:" is invalid.
+                        srm.tryGetScheduler(rev, def);
+
+                    }
+
```

Comment:
To keep the logic of putProject method simple, can you separate your validation logic from this method and copy the logic to another private method that you will create?

Reply:
Separated validation method.